bundles/persistence/org.openhab.persistence.jdbc/java/org/openhab/persistence/jdbc/internal/JdbcConfiguration.java

```diff
@@ -68,39 +70,155 @@ public JdbcConfiguration(Map<Object, Object> configuration) {
     /**
      * @{inheritDoc
      */
-    public void updateConfig(Map<Object, Object> configuration) {
+    public void updateConfig(Map<Object, Object> config) {
+        configuration = config;
+
         logger.debug("JDBC::updateConfig: configuration.size = " + configuration.size());
 
-        // Database-Url jdbc:h2:./testH2
+        // mandatory user
+        String user = (String) configuration.get("user");
+
+        // mandatory password
+        String password = (String) configuration.get("password");
+
+        // mandatory url
         String url = (String) configuration.get("url");
         Properties parsedURL = StringUtilsExt.parseJdbcURL(url);
-        logger.debug("JDBC::updateConfig: url={}", url);
-        if (StringUtils.isBlank(url) || parsedURL.getProperty("parseValid") == "false") {
-            logger.error(
-                    "JDBC::updateConfig: url The SQL database URL is missing - please configure the jdbc:url parameter in openhab.cfg");
+
+        if (user == null || StringUtils.isBlank(user)) {
+            logger.warn(
+                    "JDBC::updateConfig: SQL user is missing - please configure the jdbc:user parameter in openhab.cfg");
+        } else if (password == null || StringUtils.isBlank(password)) {
+            logger.warn("JDBC::updateConfig: SQL password is missing. Attempting to connect without password. "
+                    + "To specify a password configure the jdbc:password parameter in openhab.cfg.");
+        } else if (!(url == null || StringUtils.isBlank(url) || parsedURL.getProperty("parseValid") == "false")) {
+            logger.debug("JDBC::updateConfig: mandatory user={}", user);
+            logger.debug("JDBC::updateConfig: mandatory password=<masked> password.length={}", password.length());
+            logger.debug("JDBC::updateConfig: mandatory url={}", url);
+
+            // set database type and database type class
+            setDBDAOClass(parsedURL.getProperty("dbShortcut")); // derby, h2, hsqldb, mariadb, mysql, postgresql,
+                                                                // sqlite
+            // set user
+            dBDAO.databaseProps.setProperty("dataSource.user", user);
+
+            // set password
+            dBDAO.databaseProps.setProperty("dataSource.password", password);
+
+            // set sql-types from external config
+            setSqlTypes();
+
+            String et = (String) configuration.get("reconnectCnt");
+            if (StringUtils.isNotBlank(et)) {
+                errReconnectThreshold = Integer.parseInt(et);
```

Comment:
I mean Integer.parseInt can throw NFE, and my comment was to verify with you that you want that exception to propagate, instead of catching it and logging a warning for the user.